webapp/install_mattermost_webapp.sh

```diff
@@ -0,0 +1,26 @@
+readonly COMMITHASH=8235b5bb44
+
+echo "\n\nInstalling mattermost-webapp from the mattermost repo, using commit hash $COMMITHASH\n"
+
+if [ ! -d mattermost-webapp ]; then
+  mkdir mattermost-webapp
+fi
+
+cd mattermost-webapp
+
+if [ ! -d .git ]; then
+  git init
+  git remote add -f origin https://github.com/mattermost/mattermost.git
+  git config core.sparseCheckout true
+  echo "webapp/channels\nwebapp/platform/types\nwebapp/platform/client" >> .git/info/sparse-checkout
+fi
+
+git fetch --depth=1 origin master
+git checkout -f $COMMITHASH
```

Comment:
I am having some troubles with this because adding the remote with the `-f` flag is cloning the whole repository (not ideal) and if simply add remote and shallow clone (depth=1) then fetching the commit doesn't work (obviously).

I am wondering if we could do something like https://stackoverflow.com/a/43136160 instead.

Reply:
@streamer Thanks, can you tell me if this is better?

Reply:
Thanks @cpoile , much better. We could actually retain the sparse checkout you had before to optimize further if we want, I am not opposed to that at all.